File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.java
class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        
        int n = arrays.size();
        int ans = 0;
        int mx = arrays.get(0).get(arrays.get(0).size()-1);
        int mn = arrays.get(0).get(0);

        for(int i=1;i<n;i++){

            int a = arrays.get(i).get(0);
            int b = arrays.get(i).get(arrays.get(i).size()-1);

            ans = Math.max(ans,Math.max(Math.abs(mx-a),Math.abs(mn-b)));

            if(mx<b) mx = b;
            if(mn>a) mn = a;
        }

        return ans;
    }
}

File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays){

        int n = arrays.size();
        int ans = -1e9;
        // int n1 = arrays[0].size();
        int mx = arrays[0].back();
        int mn = arrays[0][0];

        for(int i=1;i<n;i++){

            // int n2 = arrays[i].size();
            int a = arrays[i][0];
            int b = arrays[i].back();

            ans = max(ans,max(abs(mx-a),abs(mn-b)));

            if(mx<b) mx = b;
            if(mn>a) mn = a;
        }
        
        return ans;
    }
};